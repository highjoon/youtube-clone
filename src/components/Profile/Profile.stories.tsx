import Profile from '.';

export default {
  component: Profile,
  title: 'Components/Profile',
};

export const _Profile = () => (
  <Profile>
    <Profile.Image url={'https://avatars.githubusercontent.com/u/63948484?v=4'} />
    <Profile.Nickname>HighJoon</Profile.Nickname>
  </Profile>
);
