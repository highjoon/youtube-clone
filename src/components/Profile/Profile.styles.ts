import styled from '@emotion/styled';

export const Container = styled.div`
  display: flex;
  align-items: center;
  width: 240px;
  padding: 8px 24px;
  gap: 23px;
`;

export const Image = styled.div<{ url: string }>`
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-size: contain;
  background-position: center;
  background-repeat: no-repeat;
  background-image: url(${props => props.url});
`;

export const Nickname = styled.span`
  font-size: 14px;
  font-weight: 600;
  color: var(--color-primary-black);
`;
