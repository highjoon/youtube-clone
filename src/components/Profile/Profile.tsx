import React from 'react';

import * as Styled from './Profile.styles';

interface Props {
  className?: string;
}

interface ContainerProps extends Props {
  children: React.ReactNode;
}

interface ImageProps extends Props {
  url: string;
}

interface NicknameProps extends Props {
  children: string;
}

export const Container = ({ children }: ContainerProps) => {
  return <Styled.Container>{children}</Styled.Container>;
};

export const ProfileImage = ({ url }: ImageProps) => {
  return <Styled.Image url={url} />;
};

export const Nickname = ({ children }: NicknameProps) => {
  return <Styled.Nickname>{children}</Styled.Nickname>;
};
