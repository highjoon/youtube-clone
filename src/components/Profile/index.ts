import { Container, Nickname, ProfileImage } from './Profile';

const Profile = Object.assign(Container, {
  Image: ProfileImage,
  Nickname,
});

export default Profile;
