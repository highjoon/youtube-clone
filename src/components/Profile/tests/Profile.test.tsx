import { matchers } from '@emotion/jest';
import { render } from '@testing-library/react';

import Profile from '../';

expect.extend(matchers);

const ProfileComponent = (
  <Profile>
    <Profile.Image url={'https://avatars.githubusercontent.com/u/63948484?v=4'} />
    <Profile.Nickname>HighJoon</Profile.Nickname>
  </Profile>
);

describe('Profile', () => {
  it('정상적으로 렌더링됩니다.', () => {
    const { asFragment } = render(ProfileComponent);

    expect(asFragment()).toMatchSnapshot();
  });
});
