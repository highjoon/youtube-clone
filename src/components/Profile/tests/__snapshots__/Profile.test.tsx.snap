// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile 정상적으로 렌더링됩니다. 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 240px;
  padding: 8px 24px;
  gap: 23px;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  border-radius: 50%;
  -webkit-background-size: contain;
  background-size: contain;
  -webkit-background-position: center;
  background-position: center;
  background-repeat: no-repeat;
  background-image: url(https://avatars.githubusercontent.com/u/63948484?v=4);
}

.emotion-2 {
  font-size: 14px;
  font-weight: 600;
  color: var(--color-primary-black);
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    />
    <span
      class="emotion-2"
    >
      HighJoon
    </span>
  </div>
</DocumentFragment>
`;
